docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -2008,6 +2011,159 @@ against those packages and its datatypes are not automatically converted.
+    }
+
+Because datatype ``Dep1.D`` comes from an LF 1.15 dependency, it cannot be
+upgraded, and so ``field1`` and the second element of ``field3`` can never be
+changed. However, ``field2`` and the first element of ``field3`` can be upgraded
+as new upgraded versions of ``dep2`` become available.
+
+If you compile ``main``, you can expect a warning about the use of an LF 1.15
+package's datatype in the definition of ``MyData``.
+
+.. code:: bash
+
+  > daml build
+  ...
+  warning while type checking data type Main.MyData:
+    This package has LF version 1.17, but it depends on a serializable type <dep1 package ID>:Dep1:D from package <dep 1 package ID> (dep1, 1.0.0) which has LF version 1.15.
+    
+    It is not recommended that >= LF1.17 packages depend on <= LF1.15 datatypes in places that may be serialized to the ledger, because those datatypes will not be upgradeable.
+    Upgrade this warning to an error -Werror=upgrade-serialized-non-upgradeable-dependency
+    Disable this warning entirely with -Wno-upgrade-serialized-non-upgradeable-dependency
+
+**Note:** For added safety, you may upgrade these warnings to errors with
+``-Werror=upgrade-serialized-non-upgradeable-dependency``. We recommend against disabling these
+warnings with ``-Wno-upgrade-serialized-non-upgradeable-dependency``.
+
+.. _upgrade_dont_upload_daml_script:
+
+Avoid Depending on Daml Script Packages
+---------------------------------------
+
+We recommend only depending on Daml Script packages such as ``daml-script-lts``
+in dedicated packages for running tests written in Daml Script. These packages
+should not be part of your model and should not be uploaded to the ledger.
+
+Daml Script packages are not guaranteed to be upgradeable across SDK versions.
+For this reason, if you depend on a Daml Script datatype in a serializable
+position (e.g. the field of a template), you may find your package will rely on
+a Daml Script package in a way that can neither be removed nor upgraded to the
+next SDK version. At this point, your package and any of its SCU upgrades would
+be stuck on that SDK version.
+
+For example, suppose you have a ``main`` package that depends on
+``daml-script-lts`` from SDK 2.10.0.
+
+.. code:: yaml
+
+  version: 2.10.0
+  name: main
+  ...
+  dependencies:
+  - daml-prim
+  - daml-stdlib
+  - daml-script-lts
+  build-options:
+  - --target=1.17
+
+.. code:: daml
+
+  module Main where
+
+  import qualified Daml.Script
+
+  data MyData = MyData
+    { field1 : Daml.Script.PartyIdHint
+    }
+
+Because ``MyData`` is a serializable datatype, any changes to it must be valid
+upgrade changes (e.g. adding a field) for the ``main`` package itself to be
+validly upgraded. If SDK 2.10.X were to introduce a change to
+``daml-script-lts`` that is not a valid upgrade of the ``daml-script-lts`` in
+SDK 2.10.0, then ``field1`` will not be upgradeable to the next version of the
+SDK, nor can the field be dropped because the field is used in SCU checks.
```

Comment:
```suggestion
Because ``MyData`` is a serializable datatype, any changes to it must be valid
upgrade changes (e.g. adding a field) for the ``main`` package itself to be upgraded. If SDK 2.10.X introduces a change to
``daml-script-lts`` that is not a valid upgrade of the ``daml-script-lts`` in
SDK 2.10.0, then ``field1`` is not upgradeable to the next version of the
SDK, nor can the field be dropped because the field is used in SCU checks.
```